optimum/intel/utils/modeling_utils.py

```diff
@@ -226,3 +235,199 @@ def bind_cores_for_best_perf():
             list(numa.node_to_cpus(node_id))[cpu_start : cpu_start + num_cpus_per_rank],
         )
     logger.info(f"affinity={numa.get_affinity(0)}, membind = {numa.get_membind()}")
+
+
+def _infer_library_from_model_name_or_path(
+    model_name_or_path: Union[str, Path],
+    subfolder: str = "",
+    revision: Optional[str] = None,
+    cache_dir: str = HUGGINGFACE_HUB_CACHE,
+    token: Optional[Union[bool, str]] = None,
+):
+    all_files, _ = TasksManager.get_model_files(
+        model_name_or_path, subfolder=subfolder, cache_dir=cache_dir, revision=revision, token=token
+    )
+    if "open_clip_config.json" in all_files or "open_clip_pytorch_model.bin" in all_files:
+        library_name = "open_clip"
+    else:
+        library_name = TasksManager._infer_library_from_model_name_or_path(
+            model_name_or_path=model_name_or_path, cache_dir=cache_dir
+        )
+
+    return library_name
+
+
+def _infer_library_from_model_or_model_class(
+    model: Union["PreTrainedModel", "TFPreTrainedModel", "ModelMixin", "DiffusionPipeline"]
+):
+    if model.__module__.startswith("open_clip"):
+        library_name = "open_clip"
+    elif model.__module__.startswith("optimum"):
+        # for wrapped models like timm in optimum.intel.openvino.modeling_timm
+        library_name = TasksManager._infer_library_from_model_or_model_class(model=model.model)
+    else:
+        library_name = TasksManager._infer_library_from_model_or_model_class(model=model)
+
+    return library_name
+
+
+def collect_open_clip_model_files(model_name_or_path):
+    model_files = {}
+    if os.path.isdir(model_name_or_path):
+        for filename in glob(os.path.join(model_name_or_path, "**/*"), recursive=True):
+            if filename.endswith(".bin"):
+                model_files["model"] = filename
+            if filename.endswith("open_clip_config.json"):
+                model_files["open_clip_config"] = filename
+    return model_files
+
+
+class OpenClipForZeroShotImageClassification(PreTrainedModel):
```

Comment:
why do we need OpenClipForZeroShotImageClassification / cannot load the model with  `open_clip.create_model_and_transforms` directly ?

Reply:
to be able to load models from a local path and load models without open_clip_config.json on the hugging face hub, for example laion/CLIP-convnext_base-laion400M-s13B-b51K